File: InPlan.Web/Modules/Implementacion/EstadosDetalleImplementacioneRollBack/EstadosDetalleImplementacioneRollBackForm.cs
using Serenity.ComponentModel;
using System;

namespace InPlan.Implementacion.Forms;

[FormScript("Implementacion.EstadosDetalleImplementacioneRollBack")]
[BasedOnRow(typeof(EstadosDetalleImplementacioneRollBackRow), CheckNames = true)]
public class EstadosDetalleImplementacioneRollBackForm
{
    [Tab("Datos")]
    public int IdDetalleImplementacionRollBack { get; set; }
    [HalfWidth]
    public int IdEstadoTarea { get; set; }
    [HalfWidth]
    public int IdTecnicoTarea { get; set; }
    [HalfWidth]
    public DateTime FechaEstado { get; set; }
    [HalfWidth]
    public DateTime FechaCarga { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string NotasAdicionales { get; set; }
    [TextAreaEditor(Rows = 5)]
    public string Observaciones { get; set; }
    public int UserId { get; set; }
    [Tab("Documentos")]
    public string DocumentosAdicionales { get; set; }
}

File: InPlan.Web/Modules/ServerTypes/Implementacion/EstadosDetalleImplementacioneRollBackForm.ts
import { LookupEditor, DateEditor, TextAreaEditor, IntegerEditor, MultipleImageUploadEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EstadosDetalleImplementacioneRollBackForm {
    IdDetalleImplementacionRollBack: LookupEditor;
    IdEstadoTarea: LookupEditor;
    IdTecnicoTarea: LookupEditor;
    FechaEstado: DateEditor;
    FechaCarga: DateEditor;
    NotasAdicionales: TextAreaEditor;
    Observaciones: TextAreaEditor;
    UserId: IntegerEditor;
    DocumentosAdicionales: MultipleImageUploadEditor;
}

export class EstadosDetalleImplementacioneRollBackForm extends PrefixedContext {
    static formKey = 'Implementacion.EstadosDetalleImplementacioneRollBack';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EstadosDetalleImplementacioneRollBackForm.init)  {
            EstadosDetalleImplementacioneRollBackForm.init = true;

            var w0 = LookupEditor;
            var w1 = DateEditor;
            var w2 = TextAreaEditor;
            var w3 = IntegerEditor;
            var w4 = MultipleImageUploadEditor;

            initFormType(EstadosDetalleImplementacioneRollBackForm, [
                'IdDetalleImplementacionRollBack', w0,
                'IdEstadoTarea', w0,
                'IdTecnicoTarea', w0,
                'FechaEstado', w1,
                'FechaCarga', w1,
                'NotasAdicionales', w2,
                'Observaciones', w2,
                'UserId', w3,
                'DocumentosAdicionales', w4
            ]);
        }
    }
}